applications/asset_tracker_v2/doc/cloud_module.rst

```diff
@@ -0,0 +1,209 @@
+
+To be able to use a supported cloud client library, a set of mandatory Kconfig options needs to be set.
+These typically include the cloud service hostname and the security tag associated with the certificates used to establish a connection.
+The aforementioned certificates needs to provisioned to the modem using the same security tag, prior to running the application.
+For more information on how to setup a connection and provision certificates to the modem, see the documentation for the respective client library listed in :ref:`Integration layers <integration_layers>`.
+
+.. note::
+   The :ref:`lib_nrf_cloud` library does not require any mandatory configurations to be set.
+   The nRF9160 on the DK and Thingy91 comes preprovisioned with certificates required to establish a connection to nRF Cloud.
+   The default configuration of the :ref:`lib_nrf_cloud` library uses the security tag that the nRF Cloud certificates are stored to.
+
+Configurations for AWS IoT library
+----------------------------------
+
+These options should be set in the overlay configuration file that is used to enable communication with AWS IoT, :file:`overlay-aws.conf`.
+
+* :kconfig:`CONFIG_AWS_IOT_BROKER_HOST_NAME`
+* :kconfig:`CONFIG_AWS_IOT_SEC_TAG`
+
+
+Configurations for Azure IoT Hub library
+----------------------------------------
+
+These options should be set in the overlay configuration file that is used to enable communication with Azure IoT Hub, :file:`overlay-azure.conf`.
+
+* :kconfig:`CONFIG_AZURE_IOT_HUB_DPS_HOSTNAME`
+* :kconfig:`CONFIG_AZURE_IOT_HUB_DPS_ID_SCOPE`
+* :kconfig:`CONFIG_AZURE_IOT_HUB_SEC_TAG`
+* :kconfig:`CONFIG_AZURE_FOTA_SEC_TAG`
+
+Module hierarchy
+****************
+
+The following diagram illustrates the relationship between the cloud module, integration layers, and client libraries.
+
+.. figure:: /images/asset_tracker_v2_cloud_module_hierarchy.svg
+    :alt: Cloud module hierarchy
+
+Module states
+*************
+
+The cloud module has an internal state machine with the following states:
+
+  * ``STATE_LTE_INIT`` - The initial state of the module in which it awaits the modem to be initialized.
+  * ``STATE_LTE_DISCONNECTED`` - The module has performed all required initialization and waits for the modem to connect to LTE.
+  * ``STATE_LTE_CONNECTED`` - The modem is connected to LTE and the internal cloud connection routine starts. This state has two sub-states:
+    * ``SUB_STATE_CLOUD_DISCONNECTED`` - The cloud service is disconnected.
+    * ``SUB_STATE_CLOUD_CONNECTED`` - The cloud service is connected, data can now be sent.
+  * ``STATE_SHUTDOWN`` - The module has been shut down after receiving instruction to do so from the util module.
+
+State transitions happen based on events from other modules, such as the app module, data module, and util module.
+
+Module events
+*************
+
+The :file:`asset_tracker_v2/src/events/cloud_module_event.h` header file contains a list of various events emitted by the module.
```

Comment:
Is *emitted* the right verb here? How about initiated, for example?

Reply:
The header file contains the various events that are sent out by the module. Sent is probably a better word.

Reply:
Most likely we will have to post a cleanup PR to align all modules that share common sections such as this. So that they use the same wording.